### PR TITLE
Rr vt prevent duplicates

### LINKED ISSUE #7: 6. As a user, I want to be alerted when I’m entering an item that is the same as something already on my list so I can avoid duplicate items.
AC:

* Show an error message if the user tries to submit a new item that has the exact same name as an existing item
* Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalization has been normalized and punctuation has been removed
* The user’s original input should be what gets saved in the database

### PR DESCRIPTION
**Pull request by veeteeran**
## Description

Renamed `error` variables to be more explicit. Deleted error message when user enters input. Added `propTypes` to `Error` component.

## Related Issue
Closes #7

## Acceptance Criteria

- [x] Show an error message if the user tries to submit a new item that has the exact same name as an existing item
- [x] Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalization has been normalized and punctuation has been removed
- [x] The user’s original input should be what gets saved in the database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓  | :hammer: Refactoring       |

## Testing Steps / QA Criteria

* Otherwise from your terminal, pull down this branch with `git pull origin RR-VT-prevent-duplicates` and check that branch out with `git checkout RR-VT-prevent-duplicates`
* `npm start` to launch the app.
* From add item page enter an item that already exists to see the error message
* List view reflects exact user input

### CODE REVIEW

File: src/components/AddItem.js [hunk 1]
```diff
@@ -10,12 +10,13 @@ const initialFormState = {
   lastPurchased: null,
 };
 
-const AddItem = ({ token }) => {
+const AddItem = ({ token, props }) => {
```
Comment (chrisdopuch) [inline src/components/AddItem.js:13]: @veeteeran is `props` still being used here? 
(by the way, this object you are destructuring _is_ props, this is like saying `props.props`, just FYI)

**Review by chrisdopuch — APPROVED** [review]
Nice improvement, thanks for following up!
It worked great on the demo environment. 
My only comment was the unused `props` variable, but otherwise looks excellent! 🎸

File: src/components/AddItem.js (hunk 1 above)
Comment (veeteeran) [reply src/components/AddItem.js:13]: We forgot to delete props. Definitely not supposed to be in there. Thanks Chris!

**Review by MikeBLambert — APPROVED** [review]
👍

**Pull request merged**